### PR TITLE
Bluetooth: Mesh: Limit number of Bluetooth connections used by mesh

### PR DESCRIPTION
**Pull request by PavelVPV**
When a remote devices establishes a ble connection with the mesh device
with default identity and there are more available connection slots, the
mesh stack will try to restart connectable advertisements (either
pb-gatt or proxy depending on the provisioning state). This makes
difficult for an application to advertise own connectable advertisements
as the mesh stack will not let a chance the app to start advertising.

This change adds a Kconfig option that limits number of connection slots
that the mesh stack can use for own connectable advertisements.

Additional fix:
- Ensure that another established connection with the mesh device don't
mess up pb-gatt server state.

### CODE REVIEW

File: subsys/bluetooth/mesh/pb_gatt_srv.c [hunk 1]
```diff
@@ -115,14 +115,12 @@ static void gatt_disconnected(struct bt_conn *conn, uint8_t reason)
 
 	bt_conn_get_info(conn, &info);
 	if (info.role != BT_CONN_ROLE_PERIPHERAL || !service_registered ||
-	    info.id != BT_ID_DEFAULT) {
+	    info.id != BT_ID_DEFAULT || !cli || cli->conn != conn) {
```
Comment (LingaoM) [inline subsys/bluetooth/mesh/pb_gatt_srv.c:118]: I don't want to say, the existing logic is getting harder and harder to maintain later, and refactoring needs to be considered

Comment (omkar3141) [inline subsys/bluetooth/mesh/pb_gatt_srv.c:118]: Actually, for pb_gatt there should always be exactly one connection. The provisioning process can work only with one connected client at a time. But may be this is another issue that we can fix separately.

Comment (PavelVPV) [reply subsys/bluetooth/mesh/pb_gatt_srv.c:118]: Yeah, this seems reasonable. But this specific change fixes an issue where an application uses the default identity to advertise connectable advertisements. Then once remote device connects or disconnects, this callback is triggered.

Comment (PavelVPV) [reply subsys/bluetooth/mesh/pb_gatt_srv.c:118]: Not sure how this part can be fixed. `connected` and `disconnected` callbacks are too generic.

**Review by alxelax — APPROVED** [review]

**Review by omkar3141 — APPROVED** [review]

**Review by Andrewpini — APPROVED** [review]

**Review by jhedberg — APPROVED** [review]

**Pull request merged**